R/summarystatsbayesianztest.R

```diff
@@ -0,0 +1,410 @@
+  priorSD   <- priorSD[priorSD > 0.001] # in order to keep contours from sliding into the edges
+  priorGrid <- expand.grid(mean = priorM, sd = priorSD)
+  alt       <- switch(
+    options[["hypothesis"]],
+    "notEqualToTestValue"  = "two.sided",
+    "greaterThanTestValue" = "greater",
+    "lessThanTestValue"    = "less"
+  )
+  priorGrid$bf <- .bayesianZTestsComputeBf10(x = data[["es"]], se = data[["se"]], mu = priorGrid$mean, tau = priorGrid$sd, alternative = alt)
+  priorGrid$bf[priorGrid$bf < bfLims[1]] <- bfLims[1]
+  priorGrid$bf[priorGrid$bf > bfLims[2]] <- bfLims[2]
+
+  # get plot
+  meanBreaks <- jaspGraphs::getPrettyAxisBreaks(range(priorM))
+  sdBreaks   <- jaspGraphs::getPrettyAxisBreaks(range(priorSD))
+  contours   <- bfBreaks
+
+  plot <- ggplot2::ggplot(
+    data    = priorGrid,
+    mapping = ggplot2::aes(x = sd, y = mean)
+    ) +
+    ggplot2::geom_raster(
+      mapping = ggplot2::aes(fill = bf),
+      alpha   = 0.8
+    ) +
+    ggplot2::geom_contour(
+      mapping = ggplot2::aes(z = bf),
+      breaks  = contours,
+      color   = "black", lty = 2, alpha = 0.3
+    ) +
+    metR::geom_text_contour(
+      data    = priorGrid,
+      mapping = ggplot2::aes(x = sd, y = mean, z = bf),
+      breaks  = round(contours, 2), size = 3,
+      skip = 0, parse = FALSE, label.placer = metR::label_placer_fraction(frac = 0.5)
+    ) +
+    colorspace::scale_fill_continuous_diverging(
+      palette = "Blue-Red", trans = "log",
+      mid = 0, rev = FALSE, p1 = 0.5, p2 = 1.2, ## <- tweak these values
+      breaks = bfBreaks, limits = bfLims,
+      labels = bfLabs, name = bquote(BF[10])
+    ) +
+    ggplot2::guides(fill = ggplot2::guide_colorbar(
+      barheight = 0.4,
+      barwidth  = 28,
+      label.vjust = 0.5,
+      label.hjust = 0.5)) +
+    ggplot2::scale_y_continuous(
+      name   = if (options[["dataType"]] == "esAndCiLog") gettext("exp(Prior mean)") else gettext("Prior mean"),
+      limits = range(priorM),
+      breaks = meanBreaks,
+      labels = if (options[["dataType"]] == "esAndCiLog") round(exp(meanBreaks), 2) else meanBreaks,
+      expand = c(0, 0)
+    ) +
+    ggplot2::scale_x_continuous(
+      name   = gettext("Prior standard deviation"),
+      limits = if(options[["robustnessPriorSdMin"]] == 0) c(0, max(sdBreaks)) else range(sdBreaks),
+      breaks = sdBreaks,
+      expand = c(0, 0)
+    )
+
+  plot <- plot + jaspGraphs::geom_rangeframe() + jaspGraphs::themeJaspRaw(legend.position = "top") +
+    ggplot2::theme(
+      legend.text.align = 0.5,
+      legend.title      = ggplot2::element_text(margin = ggplot2::margin(0, 10, 0, 0)),
+      legend.text       = ggplot2::element_text(size = 11),
+      plot.margin       = ggplot2::margin(t = 0, r = 50, b = 0, l = 0)
+    )
+
+  robustnessPlot[["plotObject"]] <- plot
+
+  return()
+}
+
+.bayesianZTestsComputeBf10_ <- function(x, se, mu, tau, alternative = "two.sided") {
```

Comment:
1. Why the underscore? Can you use something more informative?
2. Why do you use `tau` for prior standard deviation? (At least it seems to me that's what this argument is). This can be confusing because $\tau$ is commonly as a precision parameter.